### PR TITLE
Fix: Prettier Pre-commit staging and dist write

### LINKED ISSUE #681: Prettier should ignore `dist`
## Types of issue

- [x] Bug (High impact or critical to a production build)
- [ ] Bug (Low impact or not working as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Something else (idea, build tool, documentation improvement)

## Description

Since the introduction of `prettier` the generated dist files (including production) are being affected by it's magic. This leads to a ~5-6 increase in file size.

## Expected Behaviour and Accepted Behaviour

Prettier should ignore the dist directory.

- [ ] Should still `prettier` code and docs
- [ ] Should not do anything to generated `dist` files locally or in CI

## Steps to Reproduce (for bugs)

1. `npm run dist`
1. File size is relatively small
1. `git commit -a -m 'My distribution files'`
1. prettier runs on all files

## Screenshots

screen shot 2018-01-04 at 12 22 58

### LINKED ISSUE #682: Prettier should not stage and commit unstaged files
## Types of issue

- [ ] Bug (High impact or critical to a production build)
- [x] Bug (Low impact or not working as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Something else (idea, build tool, documentation improvement)

## Description

When committing files, prettier will run on all files regardless of git workflow state. It will then add and commit all files, even those that were originally unstaged.

## Expected Behaviour and Accepted Behaviour

Prettier shouldn't add/commit files that are unstaged.

- [ ] Should on stage and commit prettier modifications to files already staged
- [ ] Should leave unstaged files unstaged

## Current Behaviour

stages and commits everything

## Possible Solution

Could either only run prettier on staged files, and thus only commit staged files or alternatively run on all files but then only add the modifications for files that are already staged.

## Steps to Reproduce (for bugs)

1. modify two files with changes that will both fail prettier
1. stage one file (`git add file1.js`)
1. commit that one staged file (`git commit -m 'modifying one file'`)
1. prettier will run it's hooks, modify both files, stage both files, and commit both files.

## Context

Removes benefits of git staging workflows that are handy for day-to-day tasks involving refactoring.

### PR DESCRIPTION
**Pull request by omgaz**
## Description

 - git add only files that have already been staged
 - lint staged/commit to ignore dist using same mismatch pattern as the linters

## Motivation and Background Context

Fixes #681 
Fixes #682 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Manually on my local machine

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [x] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [x] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [x] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,9 +44,9 @@
     "printWidth": 120
   },
   "lint-staged": {
-    "*.{js,json,jsx}": [
+    "{src,docs,config}/**/*.{js,json,jsx}": [
       "prettier --write",
-      "git add"
+      "git update-index --again"
```
Comment (omgaz) [inline package.json]: Fixes #682

File: package.json [hunk 2]
```diff
@@ -44,9 +44,9 @@
     "printWidth": 120
   },
   "lint-staged": {
-    "*.{js,json,jsx}": [
+    "{src,docs,config}/**/*.{js,json,jsx}": [
```
Comment (omgaz) [inline package.json]: Fixes #681

File: package.json (hunk 1 above)
Comment (omgaz) [reply package.json]: [Blogs are the source of all truths](https://stacktoheap.com/blog/2016/01/02/git-add-only-files-already-staged/)

File: package.json (hunk 2 above)
Comment (omgaz) [reply package.json]: Copied from L15 & L18 for consistency.

**Comment by vinteo** [comment]
Merging...

**Pull request merged**